src/core/qgsstringutils.cpp

```diff
@@ -735,6 +735,8 @@ QString QgsStringUtils::truncateMiddleOfString( const QString &string, int maxLe
 
   // note we actually truncate an extra character, as we'll be replacing it with the ... character
   const int truncateFrom = string.length() / 2 - ( charactersToTruncate + 1 ) / 2;
+  if ( truncateFrom == 0 || charactersToTruncate >= string.length() )
```

Comment:
```suggestion
  if ( truncateFrom <= 0 )
```

It is mandatory that charactersToTruncate is lower or equal than `string.length()`